storm-core/src/jvm/backtype/storm/scheduler/resource/strategies/scheduling/DefaultResourceAwareStrategy.java

```diff
@@ -97,7 +101,7 @@ public ResourceAwareStrategy(Cluster cluster, Topologies topologies) {
 
         if (spouts.size() == 0) {
             LOG.error("Cannot find a Spout!");
-            return null;
+            return SchedulingResult.failure(SchedulingStatus.FAIL_INVALID_TOPOLOGY, "Cannot find a Spout!");
```

Comment:
If this is the case, shouldn't we have rejected the topology when it was first submitted to nimbus? It does not seem like we should have code to handle this case in the scheduler.